test/checkpoint/checkpointer.c

```diff
@@ -84,24 +84,25 @@ clock_thread(void *arg)
 
     g.ts = 0;
     while (g.running) {
-        __wt_writelock(session, &g.clock_lock);
-        ++g.ts;
-        testutil_check(
-          __wt_snprintf(buf, sizeof(buf), "oldest_timestamp=%x,stable_timestamp=%x", g.ts, g.ts));
-        testutil_check(g.conn->set_timestamp(g.conn, buf));
-        if (g.ts % 997 == 0) {
+        if (__wt_try_writelock(session, &g.clock_lock) == 0) {
```

Comment:
Why change to try lock?

Reply:
Because we now take the lock in the checkpoint thread between checkpointing and verification. During this the main thread can call join on the checkpoint thread, preventing it from releasing the clock_lock, which then causes a deadlock in the clock_lock thread as it was previously doing __wt_writelock.

Reply:
Removed the logic as its no longer needed as we don't take locks anymore in the checkpointer thread.